app/scripts/services/quota.js

```diff
@@ -260,18 +262,55 @@ angular.module("openshiftConsole")
             if (quotaKey === 'resourcequotas') {
               return false;
             }
+            if (!typesToCheck || _.includes(typesToCheck, quotaKey)) {
+              hard = usageValue(hard);
+              if (!hard) {
+                return false;
+              }
 
-            hard = usageValue(hard);
-            if (!hard) {
-              return false;
+              var used = usageValue(_.get(q, ['used', quotaKey]));
+              if (!used) {
+                return false;
+              }
+
+              return hard <= used;
             }
+          });
+        };
+        return _.some(quotas, isExceeded) || _.some(clusterQuotas, isExceeded);
+    };
 
-            var used = usageValue(_.get(q, ['used', quotaKey]));
-            if (!used) {
+    // Same as above but only looking at storage items: requests.storage, persistentvolumeclaims
+    //   Warn if you are at quota or over any storage quota for any resource.
+    var isAnyStorageQuotaExceeded = function(quotas, clusterQuotas) {
+      return isAnyQuotaExceeded(quotas, clusterQuotas, ['requests.storage', 'persistentvolumeclaims']);
+    };
+
+   // Check if requested storage for a new claim will exceed any quotas if attempted
+    var willRequestStorageQuotaBeExceeded = function(quotas, clusterQuotas, requestedStorage) {
```

Comment:
this doesnt need to be specific to storage quota, i was suggesting this be "willRequestExceedQuota", take four parameters of quotas, clusterQuotas, quotaKey, request and then we could reuse it for other stuff later.  You should really make this take request in the format that it is sent to the API which is value+unit, so you'll want to run usageValue on it.